pkg/router/routeapihelpers/validation.go

```diff
@@ -205,6 +206,11 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 			} else {
 				tlsConfig.CACertificate = string(data)
 			}
+			// HAProxy will fail to start if intermediate CA certs use unsupported signature algorithms.
+			// However, root CAs can still use unsupported algorithms since they are self-signed.
+			if err := validateCertSignatureAlgorithms(certs); err != nil {
+				result = append(result, field.Invalid(tlsFieldPath.Child("caCertificate"), "redacted ca certificate data", err.Error()))
+			}
```

Comment:
I think we also need to call `validateCertSignatureAlgorithm` for `tlsConfig.Certificate`.  If not, can you comment why not?

Reply:
Oh, I see `validateCertSignatureAlgorithm` is called by `validateCertificatePEM`.  If you make any other changes, can you add that in a comment for `tlsConfig.Certificate`?

Reply:
Yup, it's a bit tricky. I'll add a comment to call it out.